lib/galaxy/tool_util/verify/__init__.py

```diff
@@ -129,7 +130,7 @@ def get_filename(filename: str) -> str:
         # may raise an exception if `filename` does not exist (e.g. when
         # generating a tool output file from scratch with
         # `planemo test --update_test_data`).
-        if keep_outputs_dir:
+        if keep_outputs_dir and compare in ["diff", "sim_size"]:
```

Comment:
Maybe we should only do this if any of the assertion errors failed ?

I guess that even for "contains" it might be nice if you don't need to dive into the job working directory to check why it failed.

Reply:
Should we have two different flags? One to update the test data and one to just dump all the outputs when the test fails?

Reply:
Ah, no. That were the assertions. Would certainly be useful in some cases.

Reply:
I am anxious to just give up on the semantic on ``--update_test_data`` - for iterative/TDD development I think it is good to update the data even if there is not a failure or a failure is unrelated. ``--update_test_data_failed`` could be implemented though with the semantics you described and maybe a ``--dump_test_outputs`` which is unrelated and just dumps all outputs to a new directory.

I can imagine there was a time when update_test_data didn't work for work dir outputs, composite datasets, etc... and assumed Galaxy was on the same file system as the test cases... but I don't think that is the case anymore right? I think this is a useful utility that doesn't need to be buggy 😇.

Reply:
So are you arguing for not applying this change ? That sounds good.